pkg/reconciler/v1alpha1/revision/table_test.go

```diff
@@ -446,6 +446,28 @@ func TestReconcile(t *testing.T) {
 				"deploy-timeout"),
 		},
 		Key: "foo/deploy-timeout",
+	}, {
+		Name: "surface pod errors",
+		// Test the propagation of the termination state of a Pod into the revision.
+		// This initializes the world to the stable state after its first reconcile,
+		// but changes the user deployment to have a failing pod. It then verifies
+		// that Reconcile propagates this into the status of the Revision.
+		Objects: []runtime.Object{
+			rev("foo", "pod-error",
+				WithK8sServiceName, WithLogURL, AllUnknownConditions, MarkActive),
+			kpa("foo", "pod-error", WithTraffic),
+			pod("foo", "pod-error", WithFailingContainer("user-container", 5, "I failed man!")),
+			deploy("foo", "pod-error"),
+			svc("foo", "pod-error"),
+			endpoints("foo", "pod-error"),
+			image("foo", "pod-error"),
+		},
+		WantUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: rev("foo", "pod-error",
+				WithK8sServiceName, WithLogURL, AllUnknownConditions, MarkActive,
+				MarkContainerExiting(5, "I failed man!")),
```

Comment:
Cool!

Reply:
I was absolutely amazed that the test-framework already accounted for being able to list fake things with a client etc. Great stuff 😁